File: macos/src/main/resources/scala-native/Forwarders.c
#include <CoreFoundation/CFNumber.h>
#include <Security/SecItem.h>

CFStringRef SecClass()
{
    return kSecClass;
}

CFStringRef SecClassGenericPassword()
{
    return kSecClassGenericPassword;
}

CFStringRef SecAttrAccount()
{
    return kSecAttrAccount;
}

CFStringRef SecValueData()
{
    return kSecValueData;
}

CFStringRef SecReturnData()
{
    return kSecReturnData;
}

CFBooleanRef CFBooleanTrue()
{
    return kCFBooleanTrue;
}

File: macos-keychain-state-store-native/src/main/scala/macos/Forwarders.scala
package macos

import scala.scalanative.unsafe.*

// TODO: Lift these and any other common code into a shared module.
@extern
object Forwarders:

  def SecClass: macos.aliases.CFStringRef = extern

  def SecClassGenericPassword: macos.aliases.CFStringRef = extern

  def SecAttrAccount: macos.aliases.CFStringRef = extern

  def SecValueData: macos.aliases.CFStringRef = extern

  def SecReturnData: macos.aliases.CFStringRef = extern

  def CFBooleanTrue: macos.aliases.CFBooleanRef = extern
